xla/stream_executor/gpu/gpu_driver.h

```diff
@@ -132,6 +132,18 @@ class GpuDriver {
   // (supported on CUDA only)
   static void UnifiedMemoryDeallocate(GpuContext* context, void* location);
 
+  // Allocates a collective device memory space of size bytes associated with
+  // the given context via ncclMemAlloc.
+  // https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclmemalloc
+  // (supported on CUDA only)
+  static void* CollectiveMemoryAllocate(GpuContext* context, uint64_t bytes);
```

Comment:
Maybe return Status and error out on unsupported platforms? I know it's not consistent with the rest of the file, but we're trying to get there.